src/operator/single.ts

```diff
@@ -55,20 +52,25 @@ class SingleSubscriber<T> extends Subscriber<T> {
 
   protected _next(value: T): void {
     const predicate = this.predicate;
-    const currentIndex = this.index++;
-
+    this.index++;
     if (predicate) {
-      let result = tryCatch(predicate)(value, currentIndex, this.source);
-      if (result === errorObject) {
-        this.destination.error(errorObject.e);
-      } else if (result) {
-        this.applySingleValue(value);
-      }
+      this.__tryNext(value);
     } else {
       this.applySingleValue(value);
     }
   }
 
+  private __tryNext(value: T): void {
```

Comment:
do we need double underbar (_ _) here while there isn't worry to name conflict? Some of existing codebase does have it but it seems to avoid name conflict (i.e, _error and __error).